utils/utils.go

```diff
@@ -126,6 +126,52 @@ func ParseListFilters(filter []string, filterNames []string) ([]string, string)
 	return statusFilters, searchFiltersString
 }
 
+func ParseFormatParameters(filters []string) map[string]any {
+	parsedFilters := make(map[string]any)
+	for _, filter := range filters {
+		filterNameAndValue := strings.SplitN(filter, "=", 2)
+		if len(filterNameAndValue) < 2 {
+			parsedFilters[filterNameAndValue[0]] = nil
+		} else {
+			parsedFilters[filterNameAndValue[0]] = filterNameAndValue[1]
+		}
+	}
+	return parsedFilters
+}
+
+func FormatData(data *[][]any, header *[]string, formatFilters map[string]any) {
```

Comment:
I think we can make this function a little bit more readable and compact:

1. Use `row` instead of `j` and `col` instead of `i`
3. Chaining the if/else so that we do not use `continue`

Something similar to this:
```go
func FormatData(data *[][]any, header *[]string, formatFilters map[string]any) {
	if len(formatFilters) == 0 {
		return
	}

	for col := 0; col < len(*header); col++ {
		filter, include := formatFilters[(*header)[col]]

		if !include {
			// Remove the column from header and data
			...
		} else if filter != nil {
			// Remove rows not containing filter
			...
		}
	}
}
```
What do you think?

PS: note that however it is usually discouraged to modify list/array/slice while iterating over them, as it can cause many subtle bugs

Reply:
There is also another issue: when passing multiple format filters, such as `--format run_number,name`, the order in which the column headers are passed is not respected.
```console
$ reana-client list --format run_number,name
RUN_NUMBER   NAME                   
1            reana-demo-root6-roofit
$ go run . list --format run_number,name
 NAME                     RUN_NUMBER
 reana-demo-root6-roofit  1    
```

Given that this PR is already pretty big, we can address this in another issue. It is probably a good idea to do so after discussing about using external libraries to manage tables.

Reply:
Created #44